### PR TITLE
refactor: replace google.cloud.bigquery by pandas_gbq

### PR DESCRIPTION
**Pull request by rroig-at-wiris**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -21,13 +21,13 @@ classifiers=[
 [tool.poetry.dependencies]
 python = "~3.9"
 boto3 = "^1.16.50"
-google-cloud-bigquery = "^1.24.0"
```
Comment (dmas-at-wiris) [inline pyproject.toml:24]: I assume this is not required because it's installed by `pandas-gbq`, right?

File: santoku/google/bigquery.py [hunk 2]
```diff
@@ -1,85 +1,33 @@
-import pandas as pd
+from typing import Any, Dict, Union
```
Comment (dmas-at-wiris) [inline santoku/google/bigquery.py]: is `isort` working?

File: santoku/google/bigquery.py [hunk 3]
```diff
@@ -1,85 +1,33 @@
...
+    https://github.com/googleapis/python-bigquery-pandas
 
     """
 
-    def __init__(self, **kwargs) -> None:
+    def __init__(self, credential_info: Dict[str, str]) -> None:
         """
-        Base constructor. Use this to authenticate using Application Default Credentials.
-        For alternative methods of authentication use the given class methods.
+        Base constructor. Perform the authentication and the required configurations.
```
Comment (dmas-at-wiris) [inline santoku/google/bigquery.py:20]:
```suggestion
        Base constructor that performs authentication against Google Cloud Platform.
```

File: santoku/google/bigquery.py [hunk 4]
```diff
@@ -116,131 +74,15 @@ def from_aws_secrets_manager(cls, secret_name: str):
...
-            "project_id": "<project name>",
-            "private_key_id": "<private key id>",
-            "private_key": "<private key>",
-            "client_email": "<service account email>",
-            "client_id": "<client id>",
-            "auth_uri": "https://accounts.google.com/o/oauth2/auth",
-            "token_uri": "https://accounts.google.com/o/oauth2/token",
-            "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
-            "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/<email>"
-        }
```
Comment (dmas-at-wiris) [inline santoku/google/bigquery.py:135]: Maybe is it interesting to keep this?

File: santoku/google/bigquery.py [hunk 5]
```diff
@@ -250,46 +92,24 @@ def get_query_results_as_dataframe(
...
-        https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.client.Client.html#google.cloud.bigquery.client.Client.query
-        [2] :
-        https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.table.RowIterator.html#google.cloud.bigquery.table.RowIterator.to_dataframe
+        https://pandas.pydata.org/docs/reference/api/pandas.read_gbq.html
 
         Returns
         -------
         pd.DataFrame
-            The query results as a pandas dataframe
+            The query results as a Pandas DataFrame
```
Comment (dmas-at-wiris) [inline santoku/google/bigquery.py]:
```suggestion
            The query result as a Pandas DataFrame.
```

**Review by dmas-at-wiris — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (rroig-at-wiris) [reply pyproject.toml:24]: Exactly

File: santoku/google/bigquery.py (hunk 2 above)
Comment (rroig-at-wiris) [reply santoku/google/bigquery.py]: We have checked and it seems to be working fine now. Not sure why before this commit the `pandas` import was above that of `typing`.

File: santoku/google/bigquery.py (hunk 4 above)
Comment (hqiu-at-wiris) [reply santoku/google/bigquery.py:135]: Sure, we moved it to another function

**Pull request merged**